### ISSUE TITLE
`related.id` missing in graphql schema

### ISSUE THREAD

**Issue by cortopy** [VirtusLab-Open-Source/strapi-plugin-navigation#188]
I was trying to get the `id` of a related item using graphql and realised that this isn't possible.

The reason for this is that in Strapi, graphql content types follow this structure:

```
MyContentType {
   data {
      id
      attributes {
         ....
      }
   }
}
```

When the types are merged using the config in https://github.com/VirtusLab-Open-Source/strapi-plugin-navigation/blob/master/server/graphql/types/navigation-related.js#L5, only the `attributes` fields are added to the schema, leaving the id out.

I've tried to create a PR myself but I have failed because none of the approaches seemed good enough. I've tried:

1. Using [extendType](https://nexusjs.org/docs/api/extend-type) works for the navigation schema. I can add the id to all the `navigationItemRelated` set in the config. The problem with this approach is that `extendType` is polluting the schema of all objects, so the id will appear to be present as an attribute of the content type schema, when it isn't, leading to possible errors
2. Creating my own type where 

```
type NavigationRelatedData {
   id: Int!
   attributes: NavigationRelated // this is the Union type already created
}
```

This is perfectly aligned with how Strapi schemas work in v4. However, this is a breaking change and the REST responses are not structured in this manner anyway.

## Conclussion

It felt to me that finding a solution is not easy, if there is ever one. I'm happy to try to fix this, but given the change is likely to involve some compromise I wanted to see what others think first

**Comment by cyp3rius** [comment]
@ltsNotMike can you please take a look on it?

**Comment by cyp3rius** [comment]
I've analysed your approach @cortopy  and I would like to go forward with it. As for typical type resolution, you're right that `id` is missing so wrapped type just for GraphQL would be an option even as a breaking change.

Would you like to create PR or should I place is on my / any of our engineers pipeline?

**Comment by cyp3rius** [comment]
> Creating my own type where

This one @cortopy , a wrapper type just for GraphQL to support providing the id of related item

**Comment by cortopy** [comment]
so just to confirm, it's ok to change the REST responses too? I didn't find a way of "just" wrapping with a graphql type, the responses would need to change in this approach

**Comment by cyp3rius** [comment]
Let me think about that :) As I seen it the additional type for GQL didn't impact REST. But I might lost something between.

**Comment by cyp3rius** [comment]
@cortopy it has been releases as part of `v2.1.0-beta.1`

**Comment by cyp3rius** [comment]
@cortopy fix released as part of major version `2.1.0`
